### DISCUSSION TITLE
Simulation can't simuluate more than 254 inp files.ERROR 305

### DISCUSSION THREAD

**Discussion by HanchengRen** [pyswmm/pyswmm#354] [Q&A]
Hello!

I needed to run multiple INP files using Simulation of PYSWMM, but I found the following error when I ran the 254th number of files:

Exception: 
  ERROR 305: cannot open report file.

mycode:
for newpath in newpath_list:
            sim = Simulation(newfilepath)
            sim.execute()

I'm sure there's nothing wrong with the path to the file because this is the file that I generated in bulk to control the variables and the file path and address are pretty neat

I'm not a computer science major but I feel like 255 (2^8-1) is hitting some memory limits and is that manageable ?
I might need to simulate much more INP than 255 at a time

**Comment by bemcdonnell** [comment]
@rhc123123, for starters, thanks for using pyswmm!  Due the the way SWMM was developed and architected it is not possible run more than one model within one « instance » of pyswmm. Therefore, I recommend creating an asynchronous task queue (using ‘threading’) and you write workers to check out those jobs and run pyswmm_models.py using subprocess.  If I get time today I can pull together an example.

**Reply by zhangfengPL** [reply]
Hello! I suffer from the same problem, do you have a specific solution for it now?

**Comment by zhangfengPL** [comment]
Hello! I suffer from the same problem, do you have a specific solution for it now?

**Comment by dickinsonre** [comment]
I know PySWMM does not use the latest EPA SWMM5 engine but in SWMM  5.2  the number of allowed open files was 

The number of open files that SWMM can access was increased to 8192 but that applies to one model run.  The 256 sounds like an integer limit.
